src/pke/lib/schemerns/rns-cryptoparameters.cpp

```diff
@@ -426,6 +426,10 @@ std::pair<double, uint32_t> CryptoParametersRNS::EstimateLogP(uint32_t numPartQ,
             maxBits = bits;
     }
 
+    // we add an extra bit to account for for the special moduli selection logic in BGV and CKKS
+    if (addOne)
+        maxBits++;
```

Comment:
This adds at most one extra P limb by pushing the ceil of maxBits / auxBits to the next integer. This is probably enough for typical applications. But what if there is something like multiplicative depth 200, and there are more than 60 bits added because the qi's have dcrtBits+1 instead of dcrtBits bits? Shouldn't the number of buffer bits added depend on the number of qi's?

Reply:
I think this scenario is unrealistic because this would require more than 60 digits in HKS (the key would be huge). If someone uses 200 levels, they will probably use 3 digits to keep the evaluation keys comparable to ciphertexts in size.

Reply:
Actually I don't think this is possible. Max bits is found over all digits. Basically we find maxbits over all digits and then we add 1 to it.

Reply:
Makes sense.